Arrowgene.Ddon.GameServer/Quests/Quest.cs

```diff
@@ -253,4 +253,11 @@ public bool HasRewards()
             return (ItemRewards.Count > 0) || (SelectedableItemRewards.Count > 0);
         }
     }
+
+    public class QuestDoesNotExistException : Exception
```

Comment:
You could have this extend ResponseErrorException so when it throws, it returns a specific type of server error in the response packet